library/src/main/java/io/constructor/core/ConstructorIo.kt

```diff
@@ -681,18 +725,40 @@ object ConstructorIo {
      * @param orderID the identifier of the order
     */
     fun trackPurchase(customerIds: Array<String>, revenue: Double?, orderID: String, sectionName: String? = null) {
-        var completable = trackPurchaseInternal(customerIds, revenue, orderID, sectionName)
+        val items = customerIds.map { item -> PurchaseItem(item) }
+        var completable = trackPurchaseInternal(items.toTypedArray(), revenue, orderID, sectionName)
+        disposable.add(completable.subscribeOn(Schedulers.io()).subscribe({}, {
+            t -> e("Purchase error: ${t.message}")
+        }))
+    }
+
+    /**
+     * Tracks purchase events
+     * ##Example
+     * ```
+     * ConstructorIo.trackPurchase(arrayOf(PurchaseItem("1234567-AB", "1234567-AB-RED")), 25.98, "ORD-1312343")
+     * ```
+     * @param items the purchased items
+     * @param revenue the revenue of the purchase event
+     * @param orderID the identifier of the order
+     */
+    fun trackPurchase(items: Array<PurchaseItem>, revenue: Double?, orderID: String, sectionName: String? = null) {
+        var completable = trackPurchaseInternal(items, revenue, orderID, sectionName)
         disposable.add(completable.subscribeOn(Schedulers.io()).subscribe({}, {
             t -> e("Purchase error: ${t.message}")
         }))
     }
-    internal fun trackPurchaseInternal(customerIds: Array<String>, revenue: Double?, orderID: String, sectionName: String? = null): Completable {
+
+    internal fun trackPurchaseInternal(items: Array<PurchaseItem>, revenue: Double?, orderID: String, sectionName: String? = null): Completable {
+        var itemsCopy: MutableList<PurchaseItem> = ArrayList()
+        for (item in items) {
+            repeat(item.quantity ?: 1) { itemsCopy.add(item) }
```

Comment:
Yea thought that too but it complained.

<img width="585" alt="Screen Shot 2022-09-23 at 4 06 44 PM" src="https://user-images.githubusercontent.com/57369888/192068470-c2afd94c-6487-42c3-8955-4f2ba93b34ff.png">


I think it was because I still had it has an optional int? in PurchaeItem removing that seeemd to work.

Reply:
Oooo, interesting.. If we make it not optional but have it default to 1, does that require a `quantity` to be passed? 🤔

If so, we can just revert the change and keep it how you had it. 👍